gradoop-flink/src/test/java/org/gradoop/flink/model/impl/operators/cypher/capf/query/TestCAPFQuery.java

```diff
@@ -0,0 +1,500 @@
+
+    List<Long> longList = Arrays.asList(1L, 2L, 3L, 4L, 5L);
+
+    DataSet<Long> longDataSet = getExecutionEnvironment().fromCollection(longList);
+
+    tenv.registerDataSet("longs", longDataSet, "l");
+    tenv.registerFunction("GradoopId", new ZipWithGradoopId());
+
+    Table t1 = tenv.scan("longs");
+    t1 = t1.select("l, GradoopId() as id");
+
+    /*
+    TableTableSink sink = new TableTableSink(tenv);
+    t1.writeToSink(sink);
+
+    String newName = sink.getTableName();
+
+    Table t2 = tenv.scan(newName).as("l1, id1");
+    Table t3 = tenv.scan(newName).as("l2, id2");
+    */
+
+    /*
+    Table t2 = t1.as("l1, id1");
+    Table t3 = t1.as("l2, id2");
+    */
+
+    /*
+    tenv.registerTable("t1_regi", t1);
+
+    Table t2 = tenv.scan("t1_regi").as("l1, id1");
+    Table t3 = tenv.scan("t1_regi").as("l2, id2");
+
+    Table t4 = t2.join(t3).where("l1 == l2");
+
+    printTable(t4);
+  }
+
+  public static class ZipWithGradoopId extends ScalarFunction {
+
+    public GradoopId eval() {
+      return org.gradoop.common.model.impl.id.GradoopId.get();
+    }
+  }
+
+  public static class TableTableSink implements BatchTableSink<Row> {
+    private String[] fieldNames;
+    private TypeInformation[] fieldTypes;
+    private org.apache.flink.table.api.java.BatchTableEnvironment tenv;
+    private String tableName;
+
+    public TableTableSink(org.apache.flink.table.api.java.BatchTableEnvironment tenv) {
+      this.tenv = tenv;
+      this.tableName = "";
+    }
+
+    public TypeInformation<Row> getOutputType() {
+      return new RowTypeInfo(fieldTypes);
+    }
+
+    public String[] getFieldNames() {
+      String[] fieldNamesCopy = new String[fieldNames.length];
+      System.arraycopy(fieldNames, 0, fieldNamesCopy, 0, fieldNames.length);
+      return fieldNamesCopy;
+    }
+
+    public TypeInformation[] getFieldTypes() {
+      TypeInformation[] fieldTypesCopy = new TypeInformation[fieldTypes.length];
+      System.arraycopy(fieldTypes, 0, fieldTypesCopy, 0, fieldTypes.length);
+      return fieldTypesCopy;
+    }
+
+    public TableTableSink configure(String[] fieldNames, TypeInformation[] fieldTypes) {
+      String[] fieldNamesCopy = new String[fieldNames.length];
+      System.arraycopy(fieldNames, 0, fieldNamesCopy, 0, fieldNames.length);
+      this.fieldNames = fieldNamesCopy;
+      TypeInformation[] fieldTypesCopy = new TypeInformation[fieldTypes.length];
+      System.arraycopy(fieldTypes, 0, fieldTypesCopy, 0, fieldTypes.length);
+      this.fieldTypes = fieldTypesCopy;
+      return this;
+    }
+
+    @Override
+    public void emitDataSet(DataSet<Row> dataSet) {
+      this.tableName = "TableTableSink_Table_" + tenv.createUniqueTableName();
+      tenv.registerTable(this.tableName, tenv.fromDataSet(dataSet));
+    }
+
+    public String getTableName() {
+      return this.tableName;
+    }
+  }
+  */
+
+  private void printTable(Table table) {
```

Comment:
We should keep that as convenient function. Need to talk about the place for that.